src/main/java/com/aws/greengrass/logmanager/LogManagerService.java

```diff
@@ -447,37 +490,53 @@ private void loadStateFromConfiguration(String componentName) {
      */
     private void handleCloudWatchAttemptStatus(CloudWatchAttempt cloudWatchAttempt) {
         Map<String, Set<LogFile>> completedLogFilePerComponent = new ConcurrentHashMap<>();
-        Map<String, CurrentProcessingFileInformation> currentProcessingLogFilePerComponent = new HashMap<>();
 
         cloudWatchAttempt.getLogStreamUploadedSet().forEach((streamName) -> {
             CloudWatchAttemptLogInformation attemptLogInformation =
                     cloudWatchAttempt.getLogStreamsToLogEventsMap().get(streamName);
             attemptLogInformation.getAttemptLogFileInformationMap().forEach(
                     (fileHash, cloudWatchAttemptLogFileInformation) ->
                             processCloudWatchAttemptLogInformation(completedLogFilePerComponent,
-                                    currentProcessingLogFilePerComponent, attemptLogInformation, fileHash,
+                                    attemptLogInformation, fileHash,
                                     cloudWatchAttemptLogFileInformation));
         });
+
         completedLogFilePerComponent.forEach((componentName, completedFiles) -> {
             completedFiles.forEach(file -> {
                 updatelastComponentUploadedLogFile(lastComponentUploadedLogFileInstantMap, componentName,
                         file);
             });
+
             if (!componentLogConfigurations.containsKey(componentName)) {
                 return;
             }
 
             ComponentLogConfiguration componentLogConfiguration = componentLogConfigurations.get(componentName);
             completedFiles.forEach(file -> this.deleteFile(componentLogConfiguration, file));
         });
-        currentProcessingLogFilePerComponent.forEach(componentCurrentProcessingLogFile::put);
+
+
+        // Update the runtime configuration and store the last processed file information
 
         context.runOnPublishQueueAndWait(() -> {
-            componentCurrentProcessingLogFile.forEach((componentName, currentProcessingFileInformation) -> {
-                Topics componentTopics =
+            processingFilesInformation.forEach((componentName, lru) -> {
+                // Update old config value to handle downgrade from v 2.3.1 to older ones
```

Comment:
I understand why you're doing this, but boy would it be nice not to.

Was the config change from 2.2 to 2.3 backward compatible?

Reply:
We made it so that it was... we are still writing the file names into the config. I agree that this is just nasty but I don't think we have communicated to customers that downgrading from a higher to a lower version might result in issues (if we were to not handle it)